### PR TITLE
[ALLUXIO-2651] Help!!! Support Deprecated annotation in PropertyKey

### PR DESCRIPTION
**Pull request by jswudi**
https://alluxio.atlassian.net/browse/ALLUXIO-2651

### CODE REVIEW

**Comment by jswudi** [comment]
alluxio-bot, check this please

**Comment by jswudi** [comment]
alluxio-bot, check this please

**Comment by jswudi** [comment]
@apc999  Could you give some advice on how to append with a message that this property is deprecated in the toString() method?

**Comment by jswudi** [comment]
When I check with "mvn checkstyle:checkstyle", I meet a failure aobut:

Failed to execute goal on project alluxio-core-client-hdfs: Could not resolve dependencies for project org.alluxio:alluxio-core-client-hdfs:jar:1.5.0-SNAPSHOT: Failure to find org.alluxio:alluxio-core-client-fs:jar:1.5.0-SNAPSHOT in http://nexus.qmino.com/content/repositories/miredot was cached in the local repository, resolution will not be reattempted until the update interval of miredot has elapsed or updates are forced.

**Comment by jswudi** [comment]
When I check with "mvn compile findbugs:findbugs" , also meet a failure about:

Failed to execute goal com.mycila:license-maven-plugin:2.9:check (default) on project alluxio-parent: Some files do not have the expected license header

**Comment by jswudi** [comment]
I don't know how to fix them. Can anyone give some suggestions?

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 1]
```diff
@@ -1273,4 +1303,26 @@ public String getDefaultValue() {
     Object value = DEFAULT_VALUES.get(this);
     return value != null ? value.toString() : null;
   }
+
+  /**
+   * @param name the name of a property key
+   * @return if this property key is deprecated
+   */
+  public static boolean isDeprecated(String name) {
```
Comment (apc999) [inline core/common/src/main/java/alluxio/PropertyKey.java]: I feel this should take `PropertyKey` as parameter.

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 2]
```diff
@@ -1273,4 +1303,26 @@ public String getDefaultValue() {
...
+      Field field = c.getDeclaredField(name);
+      Annotation[] annotations = field.getDeclaredAnnotations();
+      for (Annotation anno : annotations) {
+        if (anno instanceof Deprecated) {
+          return true;
+        }
+      }
+      return false;
+    } catch (NoSuchFieldException e) {
+      System.err.println(String.format("No Such Field Exception: %s", e.getMessage()));
```
Comment (apc999) [inline core/common/src/main/java/alluxio/PropertyKey.java]: typically we don't print to std out or std err. 
I think we are ok to ignore here.

**Comment by apc999** [comment]
@jswudi thanks for improving this! I left two comments. PTAL

File: core/common/src/main/java/alluxio/PropertyKey.java (hunk 1 above)
Comment (jswudi) [inline core/common/src/main/java/alluxio/PropertyKey.java]: If taking PropertyKey as parameter here, then how can I get the name of the PropertyKey, like just "CONF_DIR". The DEFAULT_KEYS_MAP can only offer the key of "alluxio.conf.dir", which does not I want.

**Pull request merged**